Brokerages/Oanda/OandaRestApiV1.cs

```diff
@@ -134,74 +134,72 @@ public override bool PlaceOrder(Order order)
                 {
-                    order.BrokerId.Add(postOrderResponse.tradeReduced.id.ToString());
+                    if (order.Type != OrderType.Market)
+                    {
+                        order.BrokerId.Add(postOrderResponse.orderOpened.id.ToString());
+                    }
                 }
-            }
 
-            if (postOrderResponse.orderOpened != null && postOrderResponse.orderOpened.id > 0)
-            {
-                if (order.Type != OrderType.Market)
+                if (postOrderResponse.tradesClosed != null && postOrderResponse.tradesClosed.Count > 0)
                 {
-                    order.BrokerId.Add(postOrderResponse.orderOpened.id.ToString());
+                    marketOrderFillQuantity += postOrderResponse.tradesClosed
+                        .Where(trade => order.Type == OrderType.Market)
+                        .Sum(trade => trade.units);
                 }
-            }
 
-            if (postOrderResponse.tradesClosed != null && postOrderResponse.tradesClosed.Count > 0)
-            {
-                marketOrderFillQuantity += postOrderResponse.tradesClosed
-                    .Where(trade => order.Type == OrderType.Market)
-                    .Sum(trade => trade.units);
-            }
-
-            // send Submitted order event
-            const int orderFee = 0;
-            order.PriceCurrency = SecurityProvider.GetSecurity(order.Symbol).SymbolProperties.QuoteCurrency;
-            OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee) { Status = OrderStatus.Submitted });
-
-            if (order.Type == OrderType.Market)
-            {
-                // if market order, also send Filled order event
-                OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee)
+                // send Submitted order event
+                const int orderFee = 0;
+                order.PriceCurrency = SecurityProvider.GetSecurity(order.Symbol).SymbolProperties.QuoteCurrency;
+                OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee) { Status = OrderStatus.Submitted });
+                if (order.Type == OrderType.Market)
                 {
-                    Status = OrderStatus.Filled,
-                    FillPrice = marketOrderFillPrice,
-                    FillQuantity = marketOrderFillQuantity * Math.Sign(order.Quantity)
-                });
+                    var remainingQuantity = Convert.ToInt32(order.AbsoluteQuantity - marketOrderFillQuantity);
+                    // set order status based on remaining quantity
+                    var status = remainingQuantity > 0 ? OrderStatus.PartiallyFilled : OrderStatus.Filled;
+                    // if market order, find fill quantity and price
+                    var marketOrderFillPrice = Convert.ToDecimal(postOrderResponse.price);
+                    // if market order, also send Filled order event
+                    OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee)
```

Comment:
`OnOrderEvent` will invoke the algorithm's `OrderEvent` event handler, and if that handler submits an order, is it possible to end up in a dead-lock? Locking over user code seems risky to me